django-stubs/views/generic/detail.pyi

```diff
@@ -1,20 +1,22 @@
-from typing import Any, Optional, Type
+from typing import Any, Optional, Type, TypeVar
 
 from django.views.generic.base import ContextMixin, TemplateResponseMixin, View
 
 from django.db import models
 from django.http import HttpRequest, HttpResponse
 
+T = TypeVar('T', bound=models.Model)
+
 class SingleObjectMixin(ContextMixin):
-    model: Type[models.Model] = ...
-    queryset: models.query.QuerySet = ...
+    model: Type[T] = ...
+    queryset: models.query.QuerySet[T] = ...
     slug_field: str = ...
     context_object_name: str = ...
     slug_url_kwarg: str = ...
     pk_url_kwarg: str = ...
     query_pk_and_slug: bool = ...
-    def get_object(self, queryset: Optional[models.query.QuerySet] = ...) -> models.Model: ...
-    def get_queryset(self) -> models.query.QuerySet: ...
+    def get_object(self, queryset: Optional[models.query.QuerySet] = ...) -> T: ...
```

Comment:
`queryset: Optional[models.query.QuerySet[T]]`

Reply:
I did this on purpose actually, and was hoping to discuss it.

Here's the Django code: https://github.com/django/django/blob/main/django/views/generic/detail.py#L30

Seeing that the **default** behavior of `get_object` is to call `get_queryset`, would it not make sense that the `queryset` optional kwarg should be `QuerySet[Any]`? Passing `QuerySet[T]` would result in the same behavior as passing nothing.

Let me know if that was unclear 🙂